### PR TITLE
Generate wrappers from proto files instead of shipping them directly

### PR DESCRIPTION
**Pull request by nakulj**
*Description of changes:*
- proto wrapper classes are now generated from a `.proto` file instead of being version controlled

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by brendan-p-lynch — DISMISSED** [review]

File: amazon-kinesis-client/src/main/proto/messages.proto [hunk 1]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto2";
```
Comment (brendan-p-lynch) [inline amazon-kinesis-client/src/main/proto/messages.proto:1]: Do we have a mechanism for keeping this in line with the kpl version? I just want to make sure it isn't duplicated somewhere else.

**Review by brendan-p-lynch — COMMENTED** [review]
Overall I like this approach but just want to make sure it is consistent with what KPL is expecting

File: amazon-kinesis-client/src/main/proto/messages.proto (hunk 1 above)
Comment (nakulj) [reply amazon-kinesis-client/src/main/proto/messages.proto:1]: Ideally, yes- proto messages would be defined in a separate repo that would then be used as a submodule here and elsewhere. But from what I've seen these definitions are duplicated across multiple repos so consolidating them should be a separate effort.

File: amazon-kinesis-client/src/main/proto/messages.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+package software.amazon.kinesis.retrieval.kpl;
+
+message Tag {
```
Comment (akidambisrinivasan) [inline amazon-kinesis-client/src/main/proto/messages.proto:4]: Is this a subset of messages kcl cares about copied from here - https://github.com/awslabs/amazon-kinesis-producer/blob/master/aws/kinesis/protobuf/messages.proto ?

Comment (nakulj) [reply amazon-kinesis-client/src/main/proto/messages.proto:4]: Yes.

Comment (nakulj) [reply amazon-kinesis-client/src/main/proto/messages.proto:4]: As I mentioned in another comment, ideally these would be in a shared repo... but this is better than the previous setup I think.

Comment (akidambisrinivasan) [reply amazon-kinesis-client/src/main/proto/messages.proto:4]: Can we please add a comment "Copied from amazon-kinesis-producer/aws/kinesis/protobuf/messages.proto with subset of messages that KCL needs" or something along those lines

Comment (nakulj) [reply amazon-kinesis-client/src/main/proto/messages.proto:7]: Done

**Review by akidambisrinivasan — APPROVED** [review]

**Pull request merged**